hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ContainerImporter.java

```diff
@@ -99,6 +123,9 @@ public void importContainer(long containerID, Path tarFilePath,
         containerSet.addContainer(container);
       }
     } finally {
+      synchronized (this) {
```

Comment:
how about using Collections.synchronizedSet() instead of two synchronized blocks for addition & removal?

Reply:
Need Atomic add kind of feature, like, check and add if not present, else return. 
With Synchronized set, Add/delete is thread safe but not not achieve above kind of logic.

Reply:
We _can_ use `synchronizedSet`, because we can get rid of the `contains()` call.   `add()` returns `true` if the item was added successfully, i.e. was not already present in the `Set`.

```
if (!importContainerProgress.add(containerID)) {
  LOG...
  throw...
}
```

In fact, we _should_ use `synchronizedSet`, since currently logging is performed while holding the lock.